packages/office-ui-fabric-react/src/components/Callout/Callout.types.ts

```diff
@@ -85,21 +85,32 @@ export interface ICalloutProps extends React.HTMLAttributes<HTMLDivElement> {
   /**
    * If true then the callout will not dismiss on scroll
    * @defaultvalue false
+   * @deprecated use preventDismissOnEvent callback instead
    */
   preventDismissOnScroll?: boolean;
 
   /**
    * If true then the callout will not dismiss on resize
    * @defaultvalue false
+   * @deprecated use preventDismissOnEvent callback instead
    */
   preventDismissOnResize?: boolean;
 
   /**
    * If true then the callout will not dismiss when it loses focus
    * @defaultvalue false
+   * @deprecated use preventDismissOnEvent callback instead
    */
   preventDismissOnLostFocus?: boolean;
 
+  /**
+   *If defined, then takes priority over preventDismissOnLostFocus, preventDismissOnResize,
+   * and preventDismissOnScroll.
+   * If it returns true, then callout will not dismiss for this event.
+   * If not defined or returns false, callout can dismiss for this event.
+   */
+  preventDismissOnEvent?: (ev: any) => boolean;
```

Comment:
Instead of any, I think we might want to more strongly type this. All of those handlers seem to use Event, I might suggest typing it like `<EventType extends Event>(ev: EventType) => boolean `

Reply:
Sorry if this sounds a bit obvious, but will that also cover things like `React.MouseEvent` and others (`React.KeyboardEvent`, `React.FocusEvent`)? Tracing the hierarchy, I could see interfaces like `React.MouseEvent` extending `SyntheticEvent` which extends `BaseSyntheticEvent<Event,..>`, but couldn't find anything of the sort `extends Event`.